Build error in CPUAffinityManager.h

## Enhancement

AppleClang 17.

```
[build] /Users/zanmato/dev/tiflash/dbms/src/Common/CPUAffinityManager.h:132:15: error: implicit instantiation of undefined template 'std::shared_ptr<DB::Logger>'
[build]   132 |     LoggerPtr log;
[build]       |   
```

--- a/dbms/src/Common/CPUAffinityManager.h
+++ b/dbms/src/Common/CPUAffinityManager.h
@@ -17,6 +17,7 @@
 #include <Common/nocopyable.h>
 #include <common/defines.h>
 
+#include <memory>
 #include <string>
 #include <thread>
 #include <unordered_map>

